packages/cli/src/index.ts

```diff
@@ -333,7 +333,7 @@ program
 
     await publish({
       packageRef,
-      signer: p.signers[0],
+      signer: p.signers[0] || p.provider,
```

Comment:
This is necessary for the case that the user does not configure a signer, so, it will still be able to run Read operations